src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,110 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
 
+    private final int initSize = 10;
+    private T[] strArray;
+    private int pointer = 0;
+
+    public ArrayList() {
+        strArray = (T[]) new Object[initSize];
+    }
+
     @Override
     public void add(T value) {
-
+        ensureCapacity();
+        strArray[pointer++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        ensureCapacity();
+        pointer++;
+        wrongIndexCheck(index);
```

Comment:
1) let's move index check to the start of the method. we don't need to resize or increase `pointer` if the index is invalid

2) the logic of index check for `add()` is different - we should allow `index == pointer`
let's create a separate method `wrongIndexCheckForAdd(index)`